src/libstd/path.rs

```diff
@@ -1257,29 +1291,32 @@ impl Path {
             prefix: prefix,
             has_physical_root: has_physical_root(self.as_u8_slice(), prefix),
             front: State::Prefix,
-            back: if has_suffix(self.as_u8_slice(), prefix) { State::Suffix }
-                  else { State::Body },
+            back: State::Body,
         }
     }
 
     /// Produce an iterator over the path's components viewed as `OsStr` slices.
+    #[stable(feature = "rust1", since = "1.0.0")]
     pub fn iter(&self) -> Iter {
         Iter { inner: self.components() }
     }
 
     /// Returns an object that implements `Display` for safely printing paths
     /// that may contain non-Unicode data.
+    #[stable(feature = "rust1", since = "1.0.0")]
```

Comment:
I think at one point you had some reservations about `.display()` do you still have those?